Yoast/wordpress-seo
# Also update auto_update settings when a new addon is installed.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where Yoast addons would not inherit the auto-update settings from Yoast SEO.
* Fixes a bug where changing the auto-update settings of another plugin could stop Yoast addons from being auto-updated.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Have only free installed.
* Activate auto updates for free.
* Install premium
* Auto updates ( stored in the `auto_update_plugins` option ) should include the newly installed addon.
* Change the auto update settings of another plugin ( not from Yoast )
* Auto updates ( stored in the `auto_update_plugins` option ) should still include the newly installed addon.


### Test instructions for QA when the code is in the RC


QA can test this PR by following these steps:

* Have only an old version free installed.
* Activate auto updates for free.
* Install an old version premium.
* Clear your transients, for example using a plugin like transients manager.
* Add `add_action( 'shutdown', 'wp_maybe_auto_update' ) ` to functions.php in your theme.
* Make a request.
* Both free and premium should be updated.
* Comment the line in functions.php
* Downgrade both free and premium again.
* Change the auto update settings of another plugin ( not from Yoast )
* Clear your transients.
* Uncomment the line
* Make any request.
* Both free and premium should be updated.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Syncing auto-update settings when a new addon is installed.

## UI changes


## Other environments


## Documentation


## Quality assurance
